src/abstractdataframe/abstractdataframe.jl

```diff
@@ -375,29 +375,30 @@ describe(df, stats = [:min, :max])
 ```
 
 """
-function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} =
-                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
-    # Check that people don't specify the wrong fields.
-    allowed_fields = [:mean, :std, :min, :q25, :median, :q75,
-                      :max, :nunique, :nmissing, :first, :last, :eltype]
-    if stats == :all
-        stats = allowed_fields
+StatsBase.describe(df::AbstractDataFrame) = describe(df, :mean, :min, :median, 
+                                                     :max, :nunique, :nmissing, 
+                                                     :eltype)
+function StatsBase.describe(df::AbstractDataFrame, stats::Union{Symbol, Pair}...)
+   
+    predefined_funs = Symbol[s for s in stats if s isa Symbol] 
+    if :all in predefined_funs
+        predefined_funs = allowed_fields
     end
 
-    if stats isa Symbol
-        if !(stats in allowed_fields)
-            allowed_msg = "\nAllowed fields are: :" * join(allowed_fields, ", :")
-            throw(ArgumentError(":$stats not allowed." * allowed_msg))
-        else
-            stats = [stats]
-        end
-    end
+    custom_funs = Pair[s for s in stats if s isa Pair]
+
+    # Get the names in the order they appear
+    ordered_names = [stat isa Symbol ? stat : stat[1] for stat in stats]
```

Comment:
this should be fixed to handle `:all` case.

Reply:
also it would be good to make sure that `allunique(ordered_names)` is `true` as it will simplify the thinking about the code later simpler (and I guess we can assume that they must be unique).

Reply:
I added a check and now throw an error.